modules/axtask/src/wait_queue.rs

```diff
@@ -73,14 +88,16 @@ impl WaitQueue {
         }
     }
 
-    /// Blocks the current task and put it into the wait queue, until other task
+    /// Blocks the current task and put it into the wait list, until other task
     /// notifies it.
     pub fn wait(&self) {
-        current_run_queue::<NoPreemptIrqSave>().blocked_resched(self.queue.lock());
-        self.cancel_events(crate::current(), false);
+        declare_current_waiter!(waiter);
```

Comment:
Why we need a macro? Is the following code OK?
```rs
let waiter = WaitTaskNode::from_current();
```

P.S. we can add methods in `WaitTaskNode` as we define it in this file.

Reply:
Done